### PR TITLE
Align Bars with annotations on matplotlib

### LINKED ISSUE #4989: ErrorBars * Bars broken on matplotlib
#### ALL software version info
holoviews: 1.14.4
jupyterlab: 2.3.1
bokeh: 2.3.2
matplotlib: 3.4.2

#### Description of expected behavior and the observed behavior

Error bars should be at the tip of the bars, not floating next to it.
Alignment is correct on bokeh but broken on matplotlib.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import holoviews as hv

hv.extension('matplotlib')
cats = ['a', 'b']
levels = [3, 4]

bars = hv.Bars(zip(cats, levels)).opts(alpha=1)
errors = hv.ErrorBars(zip(cats, levels, [0.5, 0.2], [1.5, 1]), vdims=['y', 'yerrneg', 'yerrpos'])
fig = bars * errors
fig
```

#### Screenshots or screencasts of the bug in action
With matplotlib:
image

With bokeh:
image

Note that bokeh error bar is hidden by the bar itself, but that's for another bug report.

### PR DESCRIPTION
**Pull request by douglas-raillard-arm**
Align the bars with the annotations such as ErrorBars or Labels.

Fixes #4989

### CODE REVIEW

**Comment by douglas-raillard-arm** [comment]
Not 100% sure if that fixes the issue in all paths

**Comment by philippjfr** [comment]
I agree this seems like the right fix, the only worry I have here is that users in the past may have worked around the issue and now we're breaking all their workarounds.

**Comment by douglas-raillard-arm** [comment]
Maybe an explicit changelog entry would solve that problem so that people are aware they can remove their workaround ?

File: holoviews/plotting/mpl/chart.py [hunk 1]
```diff
@@ -983,12 +983,13 @@ def _create_bars(self, axis, element, ranges, style):
...
                     # mini offset needed or else combines with non-continuous
                     goffset += width / 1000
-
-                xpos = gidx+goffset if not continuous else xvals[gidx]
-                if not continuous:
-                    xticks.append(((xpos), grp_label, yalign))
+                if continuous:
+                    xpos = xvals[gidx]
+                else:
+                    xpos = gidx+goffset+0.5
```
Comment (philippjfr) [inline holoviews/plotting/mpl/chart.py:989]:
```suggestion
                    xpos = gidx+goffset
```

**Comment by philippjfr** [comment]
Looks like this was already fixed in https://github.com/holoviz/holoviews/pull/6145

**Pull request closed without merging**